firefox_ui_tests/remote/security/test_safe_browsing_notification.py

```diff
@@ -0,0 +1,150 @@
+    NoSuchElementException,
+    StaleElementException
+)
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestSafeBrowsingNotificationBar(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.urls = [
+            # Phishing URL object = mozmill 18Feb2015
+            ("safebrowsing.notAForgeryButton.label",
+             "www.google.com/safebrowsing/report_error",
+             "https://www.itisatrap.org/firefox/its-a-trap.html"),
+            # Malware URL object  = mozmill 18Feb2015
+            ("safebrowsing.notAnAttackButton.label",
+             "www.stopbadware.org",
+             "https://www.itisatrap.org/firefox/its-an-attack.html"),
+        ]
+
+        self.prefs.set_pref('browser.safebrowsing.enabled', True)
+        self.prefs.set_pref('browser.safebrowsing.malware.enabled', True)
+
+        self.tabbar = self.browser.tabbar
+
+        # TODO? workaround for bug 1084289, mozmill: "aModule.browserWindow.maximize()
+
+        # Give the browser a little time, because SafeBrowsing.jsm takes a
+        # while between start up and adding the example urls to the db.
+        time.sleep(2)
+
+    def tearDown(self):
+        self.perms.remove('www.itisatrap.org', 'safe-browsing')
+        self.prefs.restore_all_prefs
+        try:
+            self.windows.close_all([self.browser])
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_notification_bar(self):
+        with self.marionette.using_context("content"):
+            for item in self.urls:
+                label, report_page, unsafe_page = item
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_no_phishing_button(label, report_page)
+
+                # Go back to the notification bar
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+
+                # Test the get me out of here button
+                self.check_get_me_out_of_here_button()
+
+                # Go back to the notification bar
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+
+                self.marionette.set_context('chrome')
+                self.check_x_button()
+                self.marionette.set_context('content')
+
+    def check_ignore_warning_button(self, unsafe_page):
+        ignore_warning_button = self.marionette.find_element(By.ID, "ignoreWarningButton")
+
+        # This isn't clickable by the time we get here and needs a
+        # pretty significant delay.
+        time.sleep(1)
+
+        ignore_warning_button.click()
+
+        def find_main_feature_el(mn):
+            try:
+                mn.find_element("id", "main-feature")
+                return True
+            except:
+                return False
+
+        self.wait_for_condition(find_main_feature_el)
```

Comment:
You should be able to do it like `self.wait_for_condition(expected.element_present(by, locator))`. The expected module you can find here: https://dxr.mozilla.org/mozilla-central/source/testing/marionette/client/marionette/expected.py